sensors/triggers/kafka/kafka.go

```diff
@@ -194,14 +195,22 @@ func (t *KafkaTrigger) Execute(ctx context.Context, events map[string]*v1alpha1.
 		pk = trigger.URL
 	}
 
-	t.Producer.Input() <- &sarama.ProducerMessage{
+	if trigger.Partition == -1 {
```

Comment:
No, because Kafka chooses an option based on the key hash (if it's the same for all messages, all messages are delivered to the same partition). We need generate random key. You can see more information about how Kafka choosen an partition [here](https://www.confluent.io/blog/apache-kafka-producer-improvements-sticky-partitioner/)

Reply:
I believe, in line 56 (`sarama.NewConfig()`), it use a `HashPartitioner` by default (see https://github.com/Shopify/sarama/blob/eaf1def9262590b03747d463eb4e5cb287945c4d/config.go#L519) , which means if the key is `nil`, it randomly pick a partition (see https://github.com/Shopify/sarama/blob/main/partitioner.go#L158-L168), so we don't need to generate a random key.